src/vs/workbench/parts/tasks/common/problemMatcher.ts

```diff
@@ -893,16 +946,13 @@ export class ProblemPatternParser extends Parser {
 	}
 
 	private createRegularExpression(value: string): RegExp {
-		let result: RegExp | null = null;
-		if (!value) {
-			return result;
-		}
+		let result: RegExp;
 		try {
 			result = new RegExp(value);
 		} catch (err) {
 			this.error(localize('ProblemPatternParser.invalidRegexp', 'Error: The string {0} is not a valid regular expression.\n', value));
 		}
-		return result;
+		return result!;
```

Comment:
I think this is not correct (enforcing that result can't be undefined). If the value contains an invalid string creating
a RexExp will fail we catch the error and return undefined. IMO this method need to return RegExp | undefined